---
title: "ST_Immunoproteasome_Melanoma"
author: "Disha"
date: "2023-11-02"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document.

```{r}
library(hdf5r)
library(Seurat)
library(SeuratData)
library(ggplot2)
library(patchwork)
library(dplyr)
library(Matrix)
library(rhdf5)
```


```{r}
Spatial_dir <- "C:/Users/Lenovo/Desktop/Sabari_Lab/Melanoma_Spatial"
Melanoma <- Load10X_Spatial(
  data.dir=Spatial_dir,
  filename = "CytAssist_FFPE_Human_Skin_Melanoma_filtered_feature_bc_matrix.h5",
  assay = "Spatial")
```


```{r}
Melanoma_normalized <- SCTransform(Melanoma, assay = "Spatial", verbose = FALSE)
```


```{r}
#Sanity check
SpatialFeaturePlot(Melanoma_normalized, features = "S100B")
```


```{r}
# Define the features to filter
features_to_filter <- c("PSMB8", "PSMB9", "PSMB10", "CD3D", "S100B")

# Filter the Seurat spatial object based on features
filtered_Melanoma <- Melanoma_normalized[features_to_filter, ]
```


```{r}
filtered_Melanoma
```


```{r}
count_matrix <- filtered_Melanoma[["SCT"]]@data
count_matrix
```


```{r}
Melanoma_count <- t(count_matrix)
```


```{r}
Melanoma_count <- as.data.frame(Melanoma_count)
```


```{r}
Melanoma_count
```
```{r}
Melanoma_count$Immunoproteasome <- rowMeans(Melanoma_count[, c('PSMB8', 'PSMB9', 'PSMB10')], na.rm = TRUE)
Melanoma_count
```


```{r}
Melanoma_count <- Melanoma_count[ ,c("CD3D", "Immunoproteasome", "S100B"), drop = FALSE]
Melanoma_count <- t(Melanoma_count)
#converting the immunoproteasome dataset into a sparse matrix
ipsparse <- as(Melanoma_count, "sparseMatrix")
ipsparse <- as(ipsparse, "TsparseMatrix")
ipsparse
```


```{r}
library(DropletUtils)
gene.id <- rownames(ipsparse)
spatialIPhdf5_dir <- "C:/Users/Lenovo/Desktop/Sabari_Lab/Melanoma_Spatial/hdf5/IPfinal.h5"
```


```{r}
#creating a file in the format created by the CellRanger software suite

write10xCounts(
  spatialIPhdf5_dir,
  ipsparse,
  barcodes = colnames(ipsparse),
  gene.id = rownames(ipsparse),
  gene.symbol = gene.id,
  gene.type = "Gene Expression",
  overwrite = FALSE,
  type =  "HDF5",
  genome = "unknown",
  version = c("2", "3"),
  chemistry = "Single Cell 2' 2",
  original.gem.groups = 1L,
  library.ids = "custom"
)
```

```{r}
Spatial_dir <- "C:/Users/Lenovo/Desktop/Sabari_Lab/Spatial_Data_Analysis/Melanoma_Spatial/hdf5"
MelanomaIP <- Load10X_Spatial(
  data.dir=Spatial_dir,
  filename = "IPfinal.h5",
  assay = "Spatial")
```


```{r}
SpatialFeaturePlot(MelanomaIP, features = c("S100B", "Immunoproteasome",  "CD3D"))
```

